### PR TITLE
[FIX] Add missing .png to clipboard uploaded file name

### LINKED ISSUE #20165: [BUG] Add .png extension to pictures pasted from clipboard
### Description:

Rocket Chat have a great feature to paste picture from clipboard into any chat, and that is really awesome!
But name for that pasted picture generates without any extension, so if user try to save it, it will not be saved as picture. So, just add ".png" to generated name and that problem will be solved in moment.

### Steps to reproduce:

1. Get a screenshot in clipboard (with WIN+Shift+S on Windows 10)
2. Paste with CTRL+V in any Rocket.chat chat
3. Try to save that picture pressing little blue cloud icon
4. That file saved not as graphical file

### Expected behavior:

Pasted images saved as PNG files

### Actual behavior:

image
image

### Server Setup Information:

- Version of Rocket.Chat Server: 3.7.1
- Operating System: CentOS7
- Deployment Method: docker
- Number of Running Instances: 25
- DB Replicaset Oplog: Enabled
- NodeJS Version: v12.18.4
- MongoDB Version: 4.0.17

### Client Setup Information

- Desktop App or Browser Version: 3.1.1 and Chrome 87
- Operating System: Windows 10

### Additional context

Easy to fix

### Relevant logs:

No

### PR DESCRIPTION
**Pull request by dougfabris**
## Issue(s)

Closes #20165

### CODE REVIEW

File: app/ui-message/client/messageBox/messageBox.js [hunk 1]
```diff
@@ -400,11 +400,15 @@ Template.messageBox.events({
...
 		const files = items
 			.filter((item) => item.kind === 'file' && item.type.indexOf('image/') !== -1)
-			.map((item) => ({
-				file: item.getAsFile(),
-				name: `Clipboard - ${ moment().format(settings.get('Message_TimeAndDateFormat')) }`,
-			}))
-			.filter(({ file }) => file !== null);
+			.map((item) => {
+				const fileItem = item.getAsFile();
+				const fileFormat = fileItem.type.split('/')[1];
```
Comment (tassoevan) [inline app/ui-message/client/messageBox/messageBox.js:405]: SVG has the MIME type of `image/svg+xml`, resulting in a wrong file extension.

**Review by tassoevan — CHANGES_REQUESTED** [review]

**Comment by tassoevan** [comment]
There are several `image/*` formats that are not portable across browsers. Maybe that's the initial opportunity to improve our format detection. Consider the tables here: https://developer.mozilla.org/en-US/docs/Web/Media/Formats/Image_types

File: app/ui-message/client/messageBox/messageBox.js [hunk 2]
```diff
@@ -400,11 +401,14 @@ Template.messageBox.events({
...
-				file: item.getAsFile(),
-				name: `Clipboard - ${ moment().format(settings.get('Message_TimeAndDateFormat')) }`,
-			}))
-			.filter(({ file }) => file !== null);
+			.map((item) => {
+				const fileItem = item.getAsFile();
+
+				return {
+					file: fileItem,
+					name: `Clipboard - ${ moment().format(settings.get('Message_TimeAndDateFormat')) }.${ getImageExtensionFromMime(fileItem.type) }`,
```
Comment (tassoevan) [inline app/ui-message/client/messageBox/messageBox.js:409]: It's missing the case where `getImageExtensionFromMime(fileItem.type)` might be `undefined`.

File: lib/getImageExtensionFromMime.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+	'image/apng': 'apng',
+	'image/avif': 'avif',
+	'image/gif': 'gif',
+	'image/jpeg': 'jpg',
+	'image/png': 'png',
+	'image/svg+xml': 'svg',
+	'image/webp': 'webp',
+	'image/bmp': 'bmp',
+	'image/x-icon': 'ico',
+	'image/tiff': 'tif',
```
Comment (tassoevan) [inline lib/getImageExtensionFromMime.ts:11]: I'd remove these as they support on every browser is not guaranteed.

**Review by tassoevan — CHANGES_REQUESTED** [review]

File: lib/getImageExtensionFromMime.ts (hunk 3 above)
Comment (gabriellsh) [reply lib/getImageExtensionFromMime.ts:11]: As we discussed privately, I'll keep these since they're not related to the image previews.

**Review by tassoevan — APPROVED** [review]

**Pull request merged**